rasa/nlu/data_augmentation/augmenter.py

```diff
@@ -0,0 +1,699 @@
+        pooled_intents: The intents that have been selected for data augmentation.
+        output_directory: Directory to store the output reports in.
+
+    Returns:
+        Tuple representing the data augmentation summary as well as the set of changed intents.
+    """
+
+    # Retrieve intents for which performance has changed
+    changed_intents = (
+        _get_intents_with_performance_changes(
+            classification_report, intent_report, training_intents,
+        )
+        - pooled_intents
+    )
+
+    # Create and update result reports
+    report_tuple = _create_augmentation_summary(
+        pooled_intents, changed_intents, classification_report, intent_report,
+    )
+
+    intent_summary = report_tuple[0]
+    intent_report.update(report_tuple[1])
+
+    # Store report to file
+    rasa.shared.utils.io.dump_obj_as_json_to_file(
+        os.path.join(output_directory, "intent_report.json"), intent_report,
+    )
+
+    return intent_summary
+
+
+def _plot_summary_report(
+    intent_summary: Dict[Text, Dict[Text, float]], output_directory: Text,
+) -> None:
+    """Plots the data augmentation summary.
+
+    Args:
+        intent_summary: Summary report of the effect of data augmentation.
+        output_directory: Directory to store the summary plot in.
+    """
+    for metric in ["precision", "recall", "f1-score"]:
+        output_file = os.path.join(output_directory, f"{metric}_changes.png")
+        rasa.utils.plotting.plot_intent_augmentation_summary(
+            augmentation_summary=intent_summary, metric=metric, output_file=output_file,
+        )
+
+
+def _get_intents_with_performance_changes(
+    classification_report: Dict[Text, Dict[Text, Any]],
+    intent_report: Dict[Text, float],
+    all_intents: List[Text],
+    significant_figures: int = 2,
+) -> Set[Text]:
+    """
+    Extracts the intents whose performance has changed.
+
+    Args:
+        classification_report: Classification report of the model run *without* data augmentation.
+        intent_report: Report of the model run *with* data augmentation.
+        all_intents: List of all intents.
+        significant_figures: Significant figures to be taken into account when assessing whether the performance of an
+            intent has changed.
+
+    Returns:
+        Set of intents that have changed - i.e. that were affected by data augmentation.
+    """
+    changed_intents = set()
+    for intent_key in all_intents:
+        for metric in ["precision", "recall", "f1-score"]:
+            rounded_original = round(
+                classification_report[intent_key][metric], significant_figures
+            )
+            rounded_augmented = round(
+                intent_report[intent_key][metric], significant_figures
+            )
+            if rounded_original != rounded_augmented:
+                changed_intents.add(intent_key)
+
+    return changed_intents
+
+
+def _run_data_augmentation(
```

Comment:
Isn't this function more about running the `train/test cycle` and not doing the augmentation? The augmentation is already done before this function is called. I'd suggest renaming the function to `_evaluate_data_augmentation`

Reply:
Modify the function description as well accordingly